packages/editor-ui/src/components/Node/NodeCreator/Panel/NodesListPanel.vue

```diff
@@ -0,0 +1,257 @@
+const { setActiveItemIndex, attachKeydownEvent, detachKeydownEvent } = useKeyboardNavigation();
+
+const activeViewStack = computed(() => useViewStacks().activeViewStack);
+
+const viewStacks = computed(() => useViewStacks().viewStacks);
+
+const isActionsMode = computed(() => useViewStacks().activeViewStackMode === 'actions');
+const searchPlaceholder = computed(() =>
+	isActionsMode.value
+		? instance?.proxy?.$locale.baseText('nodeCreator.actionsCategory.searchActions', {
+				interpolate: { node: activeViewStack.value.title },
+		  })
+		: instance?.proxy?.$locale.baseText('nodeCreator.searchBar.searchNodes'),
+);
+
+const nodeCreatorView = computed(() => useNodeCreatorStore().selectedView);
+
+function onSearch(value: string) {
+	if (activeViewStack.value.uuid) {
+		updateCurrentViewStack({ search: value });
+		setActiveItemIndex(activeViewStack.value.activeIndex ?? 0);
+	}
+}
+
+function onTransitionEnd() {
+	setActiveItemIndex(activeViewStack.value.activeIndex ?? 0);
+}
+
+onMounted(() => {
+	attachKeydownEvent();
+	setActiveItemIndex(activeViewStack.value.activeIndex ?? 0);
+});
+
+onUnmounted(() => {
+	detachKeydownEvent();
+});
+
+watch(
+	() => nodeCreatorView.value,
+	(selectedView) => {
+		const view =
+			selectedView === TRIGGER_NODE_CREATOR_VIEW
+				? TriggerView(instance?.proxy?.$locale)
+				: RegularView(instance?.proxy?.$locale);
+
+		pushViewStack({
+			title: view.title,
+			subtitle: view?.subtitle ?? '',
+			items: view.items as INodeCreateElement[],
+			hasHeaderBg: false,
+			hasSearch: true,
+			mode: 'nodes',
+			rootView: selectedView,
+			// Root search should include all nodes
+			searchItems: mergedNodes,
+		});
+	},
+	{ immediate: true },
+);
+
+function onBackButton() {
+	popViewStack();
+}
+</script>
+
+<template>
+	<transition
+		v-if="viewStacks.length > 0"
+		:name="`panel-slide-${activeViewStack.transitionDirection}`"
+		@afterLeave="onTransitionEnd"
+	>
+		<aside :class="$style.nodesListPanel" @keydown.capture.stop :key="`${activeViewStack.uuid}`">
+			<header
+				:class="{ [$style.header]: true, [$style.hasBg]: activeViewStack.hasHeaderBg }"
+				data-test-id="nodes-list-header"
+			>
+				<div :class="$style.top">
+					<button :class="$style.backButton" @click="onBackButton" v-if="viewStacks.length > 1">
+						<font-awesome-icon :class="$style.backButtonIcon" icon="arrow-left" size="2x" />
+					</button>
+					<n8n-node-icon
+						v-if="activeViewStack.nodeIcon"
+						:class="$style.nodeIcon"
+						:type="activeViewStack.nodeIcon.iconType || 'unknown'"
+						:src="activeViewStack.nodeIcon.icon"
+						:name="activeViewStack.nodeIcon.icon"
+						:color="activeViewStack.nodeIcon.color"
+						:circle="false"
+						:showTooltip="false"
+						:size="16"
+					/>
+					<p :class="$style.title" v-text="activeViewStack.title" v-if="activeViewStack.title" />
+				</div>
+				<p
+					v-if="activeViewStack.subtitle && !activeViewStack.hasHeaderBg"
```

Comment:
do we need `hasHeaderBg`? could we just not pass `subtitle`?

Reply:
Fair point, I've removed it and changed the `hasBg` classs condition to `!subtitle`.